avrohugger-core/src/main/scala/Generator.scala

```diff
@@ -41,58 +42,65 @@ class Generator(format: SourceFormat,
   //////////////// methods for writing definitions out to file /////////////////
   def schemaToFile(
     schema: Schema,
-    outDir: String = defaultOutputDir): Unit = {
+    outDir: String = defaultOutputDir,
+    restrictedFields: Boolean = false): Unit = {
```

Comment:
I'll definitely add tests, just wanted to get quick feedback if this is the right way to go.